File: src/main/java/com/summer/lijiahao/devconfig/action/button/ApplyAction.java
package com.summer.lijiahao.devconfig.action.button;

import com.summer.lijiahao.abs.AbstractButtonAction;
import com.summer.lijiahao.abs.AbstractDialog;
import com.summer.lijiahao.base.BusinessException;
import com.summer.lijiahao.devconfig.DevConfigDialog;
import com.summer.lijiahao.devconfig.util.DataSourceUtil;
import com.summer.lijiahao.devconfig.util.TableModelUtil;

import java.awt.event.ActionEvent;

/**
 * 数据源保存
 */
public class ApplyAction extends AbstractButtonAction {
    public ApplyAction(AbstractDialog dialog) {
        super(dialog);
    }

    @Override
    public void doAction(ActionEvent event) throws BusinessException {
        DevConfigDialog dialog = (DevConfigDialog) getDialog();

        //数据源保存
        if (dialog.getTabIndex() == 0) {
            DataSourceUtil.syncCurrDataSourceValue(dialog);
        }

        //模块选择保存
        if (dialog.getTabIndex() == 1) {
            TableModelUtil.saveModuleConfig(getDialog());
        }
    }
}

File: src/main/kotlin/com/summer/lijiahao/nccdevtool/devconfig/ui/ApplyAction.kt
package com.summer.lijiahao.nccdevtool.devconfig.ui

import com.summer.lijiahao.nccdevtool.devconfig.service.NCCloudEnvSettingService
import com.summer.lijiahao.nccdevtool.devconfig.util.datasource.DataSourceUtil
import com.summer.lijiahao.nccdevtool.devconfig.util.libraries.LibrariesUtil
import com.summer.lijiahao.nccdevtool.devconfig.util.modules.ModulesUtil
import java.awt.event.ActionEvent
import javax.swing.AbstractAction

class ApplyAction(dialog: NCCloudDevConfigDialog) : AbstractAction("Apply") {

    val dialog: NCCloudDevConfigDialog

    init {
        this.dialog = dialog
    }


    override fun actionPerformed(e: ActionEvent) {
        val homePath = dialog.homeText?.text
        if (!homePath.isNullOrEmpty()) {
            //判断当前的home路径是否和环境的一致
            val homeChanged = NCCloudEnvSettingService.getInstance(dialog.event).ncHomePath != homePath
            if (homeChanged) {
                NCCloudEnvSettingService.getInstance(dialog.event).ncHomePath = homePath
                LibrariesUtil.setLibrariesAndTip(dialog.event, homePath)
            }
            DataSourceUtil.writeDataSource(dialog)
            ModulesUtil.writeModuleToConfig(dialog)

        }
    }
}